### PR TITLE
sratoolkit: import from homebrew/science

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Migrate from Homebrew/science.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/sratoolkit.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--with-ngs-sdk-prefix=#{prefix}",
+      "--with-ncbi-vdb-sources=#{buildpath}/ncbi-vdb",
+      "--with-ncbi-vdb-build=#{buildpath}/ncbi-vdb-build",
+      "--build=#{buildpath}/sra-tools-build"
+
+    system "make", "VDB_SRCDIR=#{buildpath}/ncbi-vdb", "install"
+
+    rm "#{bin}/magic"
```
Comment (ilovezfs) [inline Formula/sratoolkit.rb]: why are these rm's needed?

Comment (sjackman) [reply Formula/sratoolkit.rb]: `magic` is a non-executable ASCII file in the format of `libmagic` and `file-formula`. It's not needed by `sratoolkit`.

File: Formula/sratoolkit.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    rm_rf "#{bin}/ncbi"
```
Comment (sjackman) [inline Formula/sratoolkit.rb]:
```
bin/ncbi/certs.kfg
bin/ncbi/default.kfg
bin/ncbi/vdb-copy.kfg
```
Non-executable ASCII files not needed by `sratoolkit`.
https://gist.github.com/773ed0f9891db016913e07c5c2688d36

File: Formula/sratoolkit.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    rm_rf "#{lib}64"
```
Comment (sjackman) [inline Formula/sratoolkit.rb]: Various libraries not needed by `sratoolkit`.
```
libncbi-ngs-c++-static.a  libncbi-vdb.dylib         libngs-adapt-c++.a.1
libncbi-ngs-c++.a         libncbi-wvdb-static.a     libngs-adapt-c++.a.1.3.0
libncbi-ngs-c++.a.2       libncbi-wvdb.2.8.2.dylib  libngs-c++-static.a
libncbi-ngs-c++.a.2.8.2   libncbi-wvdb.2.dylib      libngs-c++.a
libncbi-vdb-static.a      libncbi-wvdb.a            libngs-c++.a.1
libncbi-vdb.2.8.2.dylib   libncbi-wvdb.a.2          libngs-c++.a.1.3.0
libncbi-vdb.2.dylib       libncbi-wvdb.a.2.8.2      libngs-sdk.1.3.0.dylib
libncbi-vdb.a             libncbi-wvdb.dylib        libngs-sdk.1.dylib
libncbi-vdb.a.2           libngs-adapt-c++-static.a libngs-sdk.dylib
libncbi-vdb.a.2.8.2       libngs-adapt-c++.a
```

File: Formula/sratoolkit.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    rm_rf include
```
Comment (sjackman) [inline Formula/sratoolkit.rb]: Remove header files not needed at run time by `sratoolkit`.

**Comment by ilovezfs** [comment]
If they're not needed, it's weird that upstream is installing them. It would be good to request a more restricted install target so this isn't a guessing game where we override upstream's decisions about what needs and doesn't need to be installed.

File: Formula/sratoolkit.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      end
+    end
+
+    (buildpath/"ncbi-vdb").install resource("ncbi-vdb")
+    cd "ncbi-vdb" do
+      system "./configure",
+        "--prefix=#{prefix}",
+        "--with-ngs-sdk-prefix=#{prefix}",
+        "--build=#{buildpath}/ncbi-vdb-build"
+      ENV.deparallelize { system "make" }
```
Comment (ilovezfs) [inline Formula/sratoolkit.rb]: ENV.deparallelize is a bug that needs to be reported and fixed upstream.

File: Formula/sratoolkit.rb [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      system "make", "install"
+    end
+
+    # Fix the error: ld: library not found for -lmagic-static
```
Comment (ilovezfs) [inline Formula/sratoolkit.rb]: This is a bug that needs to be fixed upstream.

File: Formula/sratoolkit.rb [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    inreplace "tools/copycat/Makefile", "-smagic-static", "-smagic"
+
+    # Fix the error: undefined reference to `SZ_encoder_enabled'
```
Comment (ilovezfs) [inline Formula/sratoolkit.rb]: This is a bug that needs to be fixed upstream.

**Comment by ilovezfs** [comment]
It's current form, I'm reluctant to bring this into core due to the

- parallelized build issue
- make install target that installs more than you think should be installed
- two unfixed-upstream string replaces for linkage issues

File: Formula/sratoolkit.rb (hunk 5 above)
Comment (sjackman) [reply Formula/sratoolkit.rb]: Reported upstream.

File: Formula/sratoolkit.rb (hunk 6 above)
Comment (sjackman) [reply Formula/sratoolkit.rb]: Reported upstream: https://github.com/ncbi/sra-tools/pull/105

**Comment by sjackman** [comment]
> If they're not needed, it's weird that upstream is installing them. It would be good to request a more restricted install target so this isn't a guessing game where we override upstream's decisions about what needs and doesn't need to be installed.

Reported upstream: https://github.com/ncbi/sra-tools/issues/106

**Comment by sjackman** [comment]
> parallelized build issue

Reported upstream: https://github.com/ncbi/ncbi-vdb/issues/16

Upstream author @kwrodarmer responded:

> Our make is currently recursive, and it runs parallel builds internally - you gain nothing by running the outer make in parallel.

**Comment by sjackman** [comment]
> make install target that installs more than you think should be installed

Reported upstream: https://github.com/ncbi/sra-tools/issues/106

> two unfixed-upstream string replaces for linkage issues

`-lsz` wasn't needed. My mistake.

`-lmagic-static` reported upstream: https://github.com/ncbi/sra-tools/pull/105

**Comment by sjackman** [comment]
The outstanding issues that you've identified have been reported upstream.

**Comment by ilovezfs** [comment]
looks like 2.8.2-5 is out

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks again @sjackman!

**Comment by sjackman** [comment]
Thanks for merging, Joe!

**Pull request closed without merging**